src/TestFramework/TestFramework/Assertions/CollectionAssert.cs

```diff
@@ -590,7 +590,108 @@ public static void IsNotSubsetOf([NotNull] ICollection? subset, [NotNull] IColle
     public static void AreEquivalent(
         [NotNullIfNotNull(nameof(actual))] ICollection? expected, [NotNullIfNotNull(nameof(expected))] ICollection? actual,
         string? message, params object?[]? parameters)
+        => AreEquivalent(expected, actual, EqualityComparer<object>.Default, message, parameters);
+
+    /// <summary>
+    /// Tests whether two collections contain the same elements and throws an
+    /// exception if either collection contains an element not in the other
+    /// collection.
+    /// </summary>
+    /// <typeparam name="T">
+    /// The type of values to compare.
+    /// </typeparam>
+    /// <param name="expected">
+    /// The first collection to compare. This contains the elements the test
+    /// expects.
+    /// </param>
+    /// <param name="actual">
+    /// The second collection to compare. This is the collection produced by
+    /// the code under test.
+    /// </param>
+    /// <param name="comparer">
+    /// The compare implementation to use when comparing elements of the collection.
+    /// </param>
+    /// <exception cref="AssertFailedException">
+    /// <paramref name="expected"/> and <paramref name="actual"/> nullabilities don't match,
+    /// or if any element was found in one of the collections but not the other.
+    /// </exception>
+    public static void AreEquivalent<T>(
+        [NotNullIfNotNull(nameof(actual))] ICollection? expected, [NotNullIfNotNull(nameof(expected))] ICollection? actual, [NotNull] IEqualityComparer<T>? comparer)
```

Comment:
As we introduce generics, I would be tempted to change to `IEnumerable<T>` to allow more types.